src/hotspot/cpu/riscv/c2_MacroAssembler_riscv.cpp

```diff
@@ -1458,6 +1458,108 @@ void C2_MacroAssembler::string_equals(Register a1, Register a2,
   BLOCK_COMMENT("} string_equals");
 }
 
+void C2_MacroAssembler::arrays_hashcode(Register ary, Register cnt,
+                                        Register result, Register tmp1, Register tmp2,
+                                        Register tmp3, Register tmp4, Register tmp5,
+                                        Register tmp6,
+                                        BasicType eltype)
+{
+  assert_different_registers(ary, cnt, result, tmp1, tmp2, tmp3, tmp4, tmp5, tmp6);
```

Comment:
We have two scratch registers t0 (x5) / t1 (x6) which should be considered for use in this assembler function.
These two are reserved from the register allocator and are suitable for keeping some short-lived values in the procedure.

Reply:
Done.